### PR TITLE
Initial GOPAX Source

### PR DESCRIPTION
**Pull request by Shella**
WIP. Todo: Add Error types, fix up `trading_pairs_404()` test.

### CODE REVIEW

File: src/sources/gopax.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    client::{Client, HttpConnector},
+    header, Body, Request,
+};
+use hyper_rustls::HttpsConnector;
+use serde::{de, Deserialize, Serialize};
+use std::str::FromStr;
+use thiserror::Error;
+
+/// Base URI for requests to the GOPAX API
+pub const BASE_URI_V4: &str = "https://api.gopax.co.kr";
```
Comment (tony-iqlusion) [inline src/sources/gopax.rs]:
```suggestion
pub const BASE_URI: &str = "https://api.gopax.co.kr";
```

File: src/sources/gopax.rs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        Self {
+            http_client: Client::builder()
+                .keep_alive(true)
+                .build(HttpsConnector::new()),
+        }
+    }
+
+    /// Get trading pairs
+    pub async fn trading_pairs(&self, pair: &Pair) -> Result<Quote, Error> {
+        let uri = format!("{}/trading-pairs/{}-{}/book", BASE_URI_V4, pair.0, pair.1);
```
Comment (tony-iqlusion) [inline src/sources/gopax.rs]:
```suggestion
        let uri = format!("{}/trading-pairs/{}-{}/book", BASE_URI, pair.0, pair.1);
```

File: src/sources/gopax.rs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                    .unwrap(),
+            );
+        }
+
+        let response = self.http_client.request(request).await?;
+        let status = response.status();
+        let body = hyper::body::aggregate(response.into_body()).await?;
+
+        if !status.is_success() {
+            let error_response: ErrorResponse =
```
Comment (tony-iqlusion) [inline src/sources/gopax.rs]: I think this API returns 200 on error, which is a little weird. We might need to use an enum for the response.

Comment (Shella) [reply src/sources/gopax.rs]: Added enum Response type and it worked. 🎉

**Pull request merged**